packages/flutter/lib/src/widgets/_window_positioner.dart

```diff
@@ -0,0 +1,510 @@
+    };
+  }
+
+  WindowPositionerAnchor _flipY() {
+    return switch (this) {
+      WindowPositionerAnchor.center => WindowPositionerAnchor.center,
+      WindowPositionerAnchor.top => WindowPositionerAnchor.bottom,
+      WindowPositionerAnchor.bottom => WindowPositionerAnchor.top,
+      WindowPositionerAnchor.left => WindowPositionerAnchor.left,
+      WindowPositionerAnchor.right => WindowPositionerAnchor.right,
+      WindowPositionerAnchor.topLeft => WindowPositionerAnchor.bottomLeft,
+      WindowPositionerAnchor.bottomLeft => WindowPositionerAnchor.topLeft,
+      WindowPositionerAnchor.topRight => WindowPositionerAnchor.bottomRight,
+      WindowPositionerAnchor.bottomRight => WindowPositionerAnchor.topRight,
+    };
+  }
+
+  Offset _offsetFor(Size size) {
+    return switch (this) {
+      WindowPositionerAnchor.center => Offset(-size.width / 2.0, -size.height / 2.0),
+      WindowPositionerAnchor.top => Offset(-size.width / 2.0, 0.0),
+      WindowPositionerAnchor.bottom => Offset(-size.width / 2.0, -size.height),
+      WindowPositionerAnchor.left => Offset(0.0, -size.height / 2.0),
+      WindowPositionerAnchor.right => Offset(-size.width, -size.height / 2.0),
+      WindowPositionerAnchor.topLeft => Offset.zero,
+      WindowPositionerAnchor.bottomLeft => Offset(0.0, -size.height),
+      WindowPositionerAnchor.topRight => Offset(-size.width, 0.0),
+      WindowPositionerAnchor.bottomRight => Offset(-size.width, -size.height),
+    };
+  }
+
+  Offset _anchorPositionFor(Rect rect) {
+    return switch (this) {
+      WindowPositionerAnchor.center => rect.center,
+      WindowPositionerAnchor.top => rect.topCenter,
+      WindowPositionerAnchor.bottom => rect.bottomCenter,
+      WindowPositionerAnchor.left => rect.centerLeft,
+      WindowPositionerAnchor.right => rect.centerRight,
+      WindowPositionerAnchor.topLeft => rect.topLeft,
+      WindowPositionerAnchor.bottomLeft => rect.bottomLeft,
+      WindowPositionerAnchor.topRight => rect.topRight,
+      WindowPositionerAnchor.bottomRight => rect.bottomRight,
+    };
+  }
+}
+
+/// The [WindowPositionerConstraintAdjustment] how a window will adjust
+/// its position when it would be partly constrained by the platform.
+///
+/// Whether a window is considered "constrained" is left to the platform
+/// to determine. For example, the surface may be partly outside of the
+/// area of the monitor, thus necessitating that the position of the
+/// child be adjusted until it is entirely inside the work area.
+///
+/// [flipX] and [flipY] mean reverse the anchor points and offset along an axis.
+/// [slideX] and [slideY] mean adjust the offset along an axis.
+/// [resizeX] and [resizeY] mean adjust the client window size along an axis.
+///
+/// The adjustments can be combined, according to a defined precedence:
+///
+///  1. [flipX] and [flipY]
+///  2. [slideX] and [slideY]
+///  3. [resizeX] and [resizeY]
```

Comment:
This is making me think of a transformation matrix and its translate/rotate/scale components.